examples/gtm/README.md

```diff
@@ -0,0 +1,208 @@
+  request: Request,
+  responseStatusCode: number,
+  responseHeaders: Headers,
+  remixContext: EntryContext,
++  context: AppLoadContext,
+) {
+  const {nonce, header, NonceProvider} = createContentSecurityPolicy({
+    shop: {
+      checkoutDomain: context.env.PUBLIC_CHECKOUT_DOMAIN,
+      storeDomain: context.env.PUBLIC_STORE_DOMAIN,
+    }
++    scriptSrc: [
++      "'self'",
++      'https://cdn.shopify.com',
++      'https://*.googletagmanager.com'
++    ],
++    imgSrc: [
++      "'self'",
++      'https://cdn.shopify.com',
++      'https://*.google-analytics.com',
++      'https://*.googletagmanager.com'
++    ],
++    connectSrc: [
++      "'self'",
++      'https://*.google-analytics.com',
++      'https://*.analytics.google.com',
++      'https://*.googletagmanager.com',
++    ],
+    shop: {
+      checkoutDomain: context.env.PUBLIC_CHECKOUT_DOMAIN,
+      storeDomain: context.env.PUBLIC_STORE_DOMAIN,
+    }
+  });
+
+  //...other code
+}
+```
+
+[View the complete component file](app/entry.server.tsx) to see these updates in context.
+
+### 2. Create a GoogleTagManager component to listen to analytics events
+
+```tsx
+import {useAnalytics} from '@shopify/hydrogen';
+import {useEffect} from 'react';
+
+declare global {
+  interface Window {
+    dataLayer: any[];
+  }
+}
+
+export function GoogleTagManager() {
+  const {subscribe, register} = useAnalytics();
+  const {ready} = register('Google Tag Manager');
+
+  useEffect(() => {
+    subscribe('product_viewed', () => {
+      // Triggering a custom event in GTM when a product is viewed
+      window.dataLayer.push({'event': 'viewed-product'});
+    });
+
+    ready();
+  }, []);
+
+  return null;
+}
+```
+
+[View the complete component file](app/components/GoogleTagManager.tsx) to see these updates in context.
+
+### 3. Edit the `root.tsx` layout file
+
+#### 3.1 Import the `GoogleTagManager` component
+
+```diff
++ import {GoogleTagManager} from '~/components/GoogleTagManager'
+```
+
+
+#### 3.2 Add the script snippet of your tag for `<head>` and `<body>`
+
+Make sure that `GTM-<YOUR_GTM_ID>` is updated to your GTM tag id.
+
+```diff
+export default function App() {
+  const nonce = useNonce();
+  const data = useLoaderData<typeof loader>();
+
+  return (
+    <html lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width,initial-scale=1" />
+        <Meta />
+        <Links />
++        <script nonce={nonce} dangerouslySetInnerHTML={{
++          __html: `(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
++            new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
++            j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
++            'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
++            })(window,document,'script','dataLayer','GTM-<YOUR_GTM_ID>');`,
++        }}></script>
```

Comment:
Any reason to not use the `Script` component? As is, this will cause hydration warnings, because the `nonce` is unavailable in the client. `suppressHydrationWarning` needs to be added as a prop: https://github.com/Shopify/hydrogen/blob/main/packages/hydrogen/src/csp/Script.tsx#L10